### ISSUE TITLE
Add better error for missing app key

### ISSUE THREAD

**Issue by cbetta** [Vonage/vonage-ruby-sdk#78]
I wanted to make a call today, authenticated with the API key and secret only, and when I made the call I received the following error:

```
*** TypeError Exception: no implicit conversion of nil into String
```

No line number, nothing. After much debugging I realised this method needs the App Key and password as well. A better error would have made this a lot easier.

**Comment by timcraft** [comment]
@cbetta You mean the application_id and private_key for the Voice API? Either way I agree, it would be good to have clearer errors for all the different types of credentials 👍

**Comment by cbetta** [comment]
@timcraft I meant that I __should__ have authenticated with the `application_id` and `private_key` but didn't, and that instead of getting a useful error got the error above.

**Comment by timcraft** [comment]
Added in 8ea3458 (available in v4.7.0)
